torch/optim/sgd.py

```diff
@@ -271,48 +272,50 @@ def _multi_tensor_sgd(params: List[Tensor],
     if len(params) == 0:
         return
 
-    if has_sparse_grad is None:
```

Comment:
Note: I no longer use has_sparse_grad

Reply:
But I don't think I can remove it from the signature because the current construct feeds has_sparse_grad from sgd() and expects the generic function to be able to deal with it

Reply:
And I'm not sure if we want to get rid of that generic abstraction + specialize (doesn't seem worth it)

Reply:
I think we would like to deprecate and remove this argument from the public sgd() function (I don't think there are too many users of this API so that won't be too disruptive).
Also the step() function can stop computing it and passing it to the sgd() function.

Both of these can be done in a follow-up though.